addons/website_livechat/models/mail_channel.py

```diff
@@ -30,20 +31,29 @@ def channel_info(self, extra_info=False):
         channel_infos = super(MailChannel, self).channel_info(extra_info)
         channel_infos_dict = dict((c['id'], c) for c in channel_infos)
         for channel in self:
-            visitor = channel.livechat_visitor_id
-            if visitor:
-                channel_infos_dict[channel.id]['visitor'] = {
-                    'name': visitor.display_name,
-                    'country_code': visitor.country_id.code.lower() if visitor.country_id else False,
-                    'country_id': visitor.country_id.id,
-                    'is_connected': visitor.is_connected,
-                    'history': self.sudo()._get_visitor_history(visitor),
-                    'website': visitor.website_id.name,
-                    'lang': visitor.lang_id.name,
-                    'partner_id': visitor.partner_id.id,
-                }
+            try:
+                visitor_info = channel._get_livechat_visitor_info()
+            except AccessError:
+                visitor_info = {}
+            if visitor_info:
+                channel_infos_dict[channel.id]['visitor'] = visitor_info
```

Comment:
Move this into the try block as a single line instead
We don't want to add an empty value in case of exception, because that indicates there is no visitor (when it's not actually true). It's better to not add the key to indicate there is no data.

Reply:
Done, moved. However, an empty value wasn't going to be added thanks to the conditional `if visitor_info` just above.